filament/src/components/RenderableManager.cpp

```diff
@@ -235,12 +259,121 @@ RenderableManager::Builder& RenderableManager::Builder::globalBlendOrderEnabled(
     return *this;
 }
 
+UTILS_NOINLINE
+void RenderableManager::Builder::processBoneIndicesAndWights(Engine& engine, Entity entity){
+    size_t maxPairsCount = 0; //size of texture, number of bone pairs
+    size_t maxPairsCountPerVertex = 0; //maximum of number of bone per vertex
+
+    for(auto iBonePair = mImpl->mBonePairs.begin(); iBonePair != mImpl->mBonePairs.end(); ++iBonePair){
+        auto primitiveIndex = iBonePair->first;
+        auto entries = mImpl->mEntries;
+        ASSERT_PRECONDITION(primitiveIndex < entries.size() && primitiveIndex >= 0,
+            "[primitive @ %u] primitiveindex is out of size (%u)", primitiveIndex, entries.size());
+        auto entry = mImpl->mEntries[primitiveIndex];
+        auto bonePairsForPrimitive = iBonePair->second;
+        auto vertexCount = entry.vertices->getVertexCount();
+        ASSERT_PRECONDITION(bonePairsForPrimitive.size() == vertexCount,
+            "[primitive @ %u] bone indices and weights pairs count (%u) must be equal to vertex count (%u)",
+            primitiveIndex, bonePairsForPrimitive.size(), vertexCount);
+        auto const& declaredAttributes = downcast(entry.vertices)->getDeclaredAttributes();
+        ASSERT_PRECONDITION(!declaredAttributes[VertexAttribute::BONE_INDICES],
+            "[entity=%u, primitive @ %u] vertex attribute BONE_INDICES for skinning are already defined",
+            entity.getId(), primitiveIndex);
+        ASSERT_PRECONDITION(!declaredAttributes[VertexAttribute::BONE_WEIGHTS],
+            "[entity=%u, primitive @ %u] vertex attribute BONE_WEIGHTS for skinning are already defined",
+            entity.getId(), primitiveIndex);
+        for (size_t iVertex = 0; iVertex < vertexCount; iVertex++){
+            auto bonesPerVertex = bonePairsForPrimitive[iVertex].size();
+            maxPairsCount +=  bonesPerVertex;
+            maxPairsCountPerVertex = max(bonesPerVertex, (uint) maxPairsCountPerVertex);
+        }
+    }
+
+    size_t pairsCount = 0; // counting of number of pairs stored in texture
+    float2* tempPairs; // temporary indices and weights for one vertex
+    if (maxPairsCount) { // at least one primitive has bone indices and weights
+        mImpl->mBoneIndicesAndWeight = (float2*) malloc(
+            maxPairsCount * 2 * 4); // final texture data, indices and weights
```

Comment:
Use `sizeof(float2)` to help readability.

Reply:
Fixed